src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in inventory.");
+        }
+        int newQuantity = currentQuantity - quantity;
+        if (newQuantity < 0) {
+            throw new RuntimeException("The resulting quantity of "
+                    + fruit + " cannot be negative.");
```

Comment:
This check is redundant because you've already ensured that 'currentQuantity' is greater than or equal to 'quantity'. The 'newQuantity' will never be negative after the subtraction on line 12, so you can safely remove lines 13-15.